### PR TITLE
Preserve DASC numeric argument compatibility

### PR DESCRIPTION
**Pull request by kaix-nv**
## Summary

Addresses all remaining suggestions and the CodeRabbit oversized-integer finding on #2395:

- keep one source of truth for epsilon/static-gate validation in `config.py`
- preserve duck-typed scalar compatibility instead of narrowing to exact Python float/int types
- normalize TypeError, ValueError, and OverflowError to the public ValueError contract
- cover wrong-type, non-finite, oversized-integer, and tensor-scalar cases through both exported analysis paths
- use `torch.as_tensor` for scalar epsilon conversion without copy-construction warnings

`numbers.Real` was considered, but it excludes PyTorch scalar tensors (and Decimal) on the validated host, so guarded numeric operations preserve the previous API more faithfully.

## Validation

- combined state/weight/attention sparsity suite: 321 passed, 1 optional Megatron skip
- pre-commit hooks on all three changed files
- signed commit with DCO sign-off

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved validation for analysis settings, including finite epsilon values and valid static gate inputs.
  * Ensured horizon calculations remain stable and produce finite results across supported scalar tensor inputs.
  * Improved handling of oversized numeric values and invalid tensor inputs to provide more reliable configuration behavior.

* **Tests**
  * Expanded coverage for invalid numeric values, including oversized integers and tensors containing NaN.
  * Added validation for tensor-based scalar arguments in horizon calculations.

### CODE REVIEW

File: modelopt/torch/sparsity/state_sparsity/config.py [hunk 1]
```diff
@@ -31,6 +31,28 @@
...
+        epsilon_is_valid = False
+    if not epsilon_is_valid:
+        raise ValueError("epsilon must be finite and in (0, 1)")
+
+    try:
+        static_gate_input_is_valid = math.isfinite(static_gate_input)
+    except (TypeError, ValueError, OverflowError):
+        static_gate_input_is_valid = False
+    if not static_gate_input_is_valid:
+        raise ValueError("static_gate_input must be finite")
```
[the remark this answers is not in the document]
Comment (kaix-nv) [reply modelopt/torch/sparsity/state_sparsity/config.py:65]: Fixed in #2397 with a stricter scalar boundary rather than catching arbitrary framework RuntimeError. The annotated inputs now require numbers.Real; NumPy real scalars remain accepted, while scalar and multi-element tensors are rejected uniformly with ValueError before math.isfinite. Multi-element regressions cover both public entry points.

File: tests/unit/torch/sparsity/state_sparsity/test_dasc.py [hunk 2]
```diff
@@ -489,6 +493,18 @@ def test_horizon_computation_rejects_invalid_public_arguments(argument, value, m
...
+    """Preserve support for real-like scalar values accepted by the numeric operations."""
+    horizons = mtss.compute_gdn_decay_horizons(
+        torch.tensor([0.0]),
+        torch.tensor([0.0]),
+        epsilon=torch.tensor(1e-3),  # type: ignore[arg-type]
+        static_gate_input=torch.tensor(-0.3),  # type: ignore[arg-type]
+    )
+    assert horizons.shape == (1,)
+    assert torch.isfinite(horizons).all()
+
```
[the remark this answers is not in the document]
Comment (kaix-nv) [reply tests/unit/torch/sparsity/state_sparsity/test_dasc.py]: Agreed and fixed in #2397. Tensor support was inconsistent with the public float annotations, so the tensor-scalar acceptance test and as_tensor change are removed. numbers.Real now makes the intended scalar contract explicit at runtime.
